### PR TITLE
[sentry-config] add pagination

### PR DESCRIPTION
**Pull request by maorfr**
as we grew larger in numbers, some calls now do not return all results and include a `link` header in the response indicating of more pages.
this PR adds usage of these pages to return the complete response (if pages exist, the response is a list of items).

### CODE REVIEW

File: reconcile/utils/sentry_client.py [hunk 1]
```diff
@@ -27,10 +28,35 @@ def _do_sentry_api_call_(self, method, path, slugs, payload=None):
         response.raise_for_status()
 
         try:
-            return response.json()
+            all_results = response.json()
         except json.decoder.JSONDecodeError:
             return
 
+        # there may be more pages if the response contains a link header
```
Comment (aditya-konarde) [inline reconcile/utils/sentry_client.py:35]: Thank you for adding a note here!

Comment (aditya-konarde) [reply reconcile/utils/sentry_client.py:35]: Perhaps add an example as well for easier reading?

File: reconcile/utils/sentry_client.py [hunk 2]
```diff
@@ -27,10 +28,35 @@ def _do_sentry_api_call_(self, method, path, slugs, payload=None):
...
+        while True:
+            link = response.headers.get('link')
+            if not link:
+                break
+            # 2nd item is the next page
+            next_item = link.split(', ')[1]
+            # copied with love from
+            # https://stackoverflow.com/questions/10663093/
+            # use-python-format-string-in-reverse-for-parsing
+            _, rel, results, cursor = parse.parse(item_format, next_item)
```
Comment (aditya-konarde) [inline reconcile/utils/sentry_client.py:49]: :heart_eyes:

File: reconcile/utils/sentry_client.py [hunk 3]
```diff
@@ -27,10 +28,35 @@ def _do_sentry_api_call_(self, method, path, slugs, payload=None):
...
+            if rel != 'next' or results != 'true':
+                break
+            response = \
+                call(f"{url}?&cursor={cursor}", headers=headers, json=payload)
+            response.raise_for_status()
+            # if there are pages, each response is a list to append
```
Comment (apahim) [inline reconcile/utils/sentry_client.py:55]: append -> extend

**Review by apahim — APPROVED** [review]
noice

File: reconcile/utils/sentry_client.py (hunk 1 above)
Comment (aditya-konarde) [reply reconcile/utils/sentry_client.py:35]: Even better if we add a link to the sentry API documentation here, if available

File: reconcile/utils/sentry_client.py [hunk 4]
```diff
@@ -27,10 +28,35 @@ def _do_sentry_api_call_(self, method, path, slugs, payload=None):
...
+            # 2nd item is the next page
+            next_item = link.split(', ')[1]
+            # copied with love from
+            # https://stackoverflow.com/questions/10663093/
+            # use-python-format-string-in-reverse-for-parsing
+            _, rel, results, cursor = parse.parse(item_format, next_item)
+            if rel != 'next' or results != 'true':
+                break
+            response = \
+                call(f"{url}?&cursor={cursor}", headers=headers, json=payload)
```
Comment (aditya-konarde) [inline reconcile/utils/sentry_client.py:53]: Curious: in what situations does 'call' fail? Is it worth wrapping it in a try/catch block?

**Review by aditya-konarde — APPROVED** [review]
Thanks for working on this Maor! I have added a few suggestions and questions I had. Overall LGTM, please feel free to move ahead after addressing/acknowledging the suggestions :)

File: reconcile/utils/sentry_client.py (hunk 3 above)
Comment (maorfr) [reply reconcile/utils/sentry_client.py]: thanks!

File: reconcile/utils/sentry_client.py (hunk 1 above)
Comment (maorfr) [reply reconcile/utils/sentry_client.py:35]: this is the example (right here below):
https://github.com/app-sre/qontract-reconcile/pull/1303/files#diff-c4b8fdf06e29c3116b2012bc23e99e3833efbd32539bbdd8b84ea54fc17ccf6bR36-R39

Comment (maorfr) [reply reconcile/utils/sentry_client.py:35]: the API documentation is for a newer version then the one we are using. I couldn't find this (had to dig through the response to implement this)

File: reconcile/utils/sentry_client.py (hunk 4 above)
Comment (maorfr) [reply reconcile/utils/sentry_client.py:53]: the way to test the result of call (`requests.get` for example) is with `response.raise_for_status()` (which will raise an exception we want to see if something happened).

**Pull request merged**